src/main/scala/org/broadinstitute/dsde/rawls/webservice/SubmissionApiService.scala

```diff
@@ -32,13 +34,16 @@ trait SubmissionApiService extends HttpService with PerRequestCreator {
     new ApiResponse(code = 409, message = "Method Configuration failed to resolve input expressions with the supplied Entity"),
     new ApiResponse(code = 500, message = "Rawls Internal Error")
   ))
-  def submissionRoute = cookie("iPlanetDirectoryPro") { securityTokenCookie =>
-    path("workspaces" / Segment / Segment / "submissions") { (workspaceNamespace, workspaceName) =>
-      post {
-        entity(as[SubmissionRequest]) { submission =>
-          requestContext => perRequest(requestContext,
-                                       WorkspaceService.props(workspaceServiceConstructor),
-                                       WorkspaceService.CreateSubmission(WorkspaceName(workspaceNamespace,workspaceName),submission,securityTokenCookie))
+  def submissionRoute = usernameFromCookie() { userId =>
+    // TODO: reads the cookie twice!
+    cookie("iPlanetDirectoryPro") { securityTokenCookie =>
+      path("workspaces" / Segment / Segment / "submissions") { (workspaceNamespace, workspaceName) =>
+        post {
+          entity(as[SubmissionRequest]) { submission =>
+            requestContext => perRequest(requestContext,
+              WorkspaceService.props(workspaceServiceConstructor),
+              WorkspaceService.CreateSubmission(userId, WorkspaceName(workspaceNamespace, workspaceName), submission, securityTokenCookie))
```

Comment:
you could add userId to workspaceServiceConstructor instead of to each and every message... I know you have done the work already but it is probably cleaner the other way... up to you.